pkg/webhook/validate.go

```diff
@@ -150,12 +150,25 @@ func staticValidateSriovNetworkNodePolicy(cr *sriovnetworkv1.SriovNetworkNodePol
 	// To configure RoCE on baremetal or virtual machine:
 	// BM: DeviceType = netdevice && isRdma = true
 	// VM: DeviceType = vfio-pci && isRdma = false
-	if cr.Spec.DeviceType == "vfio-pci" && cr.Spec.IsRdma {
+	if cr.Spec.DeviceType == constants.DeviceTypeVfioPci && cr.Spec.IsRdma {
 		return false, fmt.Errorf("'deviceType: vfio-pci' conflicts with 'isRdma: true'; Set 'deviceType' to (string)'netdevice' Or Set 'isRdma' to (bool)'false'")
 	}
 	if strings.EqualFold(cr.Spec.LinkType, constants.LinkTypeIB) && !cr.Spec.IsRdma {
 		return false, fmt.Errorf("'linkType: ib or IB' requires 'isRdma: true'; Set 'isRdma' to (bool)'true'")
 	}
+
+	// vdpa: deviceType must be set to 'netdevice'
+	if cr.Spec.DeviceType != constants.DeviceTypeNetDevice && cr.Spec.VdpaType == constants.VdpaTypeVirtio {
+		return false, fmt.Errorf("'deviceType: %s' conflicts with 'vdpaType: virtio'; Set 'deviceType' to (string)'netdevice' Or Remove 'vdpaType'", cr.Spec.DeviceType)
+	}
+	// vdpa: device must be configured in switchdev mode
+	if cr.Spec.VdpaType == constants.VdpaTypeVirtio && cr.Spec.EswitchMode != sriovnetworkv1.ESwithModeSwitchDev {
+		return false, fmt.Errorf("virtio/vdpa requires the device to be configured in switchdev mode")
+	}
+	// vdpa: only mellanox cards are supported
+	if cr.Spec.VdpaType == constants.VdpaTypeVirtio && cr.Spec.NicSelector.Vendor != "" && cr.Spec.NicSelector.Vendor != MellanoxID {
```

Comment:
I think is better to add the validation here https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/377/files#diff-8fd5d83d7413d77ee2eab8e5c1a2ea1623d63e2aec69017a95811db9ad9edf52R235
so we can always check it not only if vendorID is configured

Reply:
done, thx